### DISCUSSION TITLE
(not) disabling protein grouping

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#496] [Q&A]
In DIA-NN one can disable protein grouping by setting it to Off. 
In discussion #493 you have recommended to turn off protein grouping when using MSFragger libs. 
Actually, I  think it may make sense to leave protein grouping on when using MSFragger libs: 

Let's say that there are two similar proteins ProtA and ProtB which have unique peptides A and B and shared ones 'S'. In the library the depth is higher than in the actual DIA sample runs. So in the library, both A and B are detected. The S peptides should be assigned to both protein groups ProtA and ProtB of the library. 

There now may be ambiguity if protein grouping is just taken from the library. If only S peptides are identified but neither A or B, S cannot be assigned as razor peptides to ProtA or ProtB. In such case one would want to re-group the proteins, so taht ProtA and ProtB are in one protein group 'AB' for which S peptides are now specific. However, that requires re-grouping, since this is not a group in the library which only has ProtA and Protb. 

As protein grouping is depth-dependent and the depth can vary between library and actual samples, it appears to me that re-grouping makes sense in these scenarios. 
Q1: Do you agree? 
Q2: Is there another reason to disable protein grouping when using MSFragger input libraries and similar ones?
Q3: What about predicted libraries (which are very deep and have high resolution of protein groups not observed in actual samples) ? There re-grouping should make even more sense.

**Comment by vdemichev** [comment]
Q1: I would think that having grouping on/off with MSFragger libs would have very little effect - just because almost all peptides will have been matched to a single protein in the library. But, yes, maybe can keep it on.
Q2: When enabling protein grouping in DIA-NN, a FASTA needs to be supplied. And it needs to be identical to the one used to generate the library, ideally. So a bit more hassle. Btw, proteotypicity information is lost with MSFragger libs (but can be recovered from some other MSFragger output files I think), need to keep it in mind, with both grouping in DIA-NN on or off.
Q3: predicted libs from FASTA would always lack protein grouping, right? So protein grouping needs to be enabled.

**Reply by JM-Bader** [reply]
Q1: Ah ok. I completely agree, that in many cases the change caused by protein grouping is probably minor. Since you recommended to turn them off, I though thought that there is a (probably severe) reason to have them off. From what I understand here, that is not the case and it should be ok to have it off. 
Q2: You are right, a bit more hassle. 
What do you mean proteotypicity information is lost ? To me proteotypicity means 'specific to a protein group'. So if there is protein group, peptides specific to that protein group are proteotypic. 
In any way, if lost - re-grouping with DIA-NN should restore proteotypicity. Doesn't it? Likewise, I expect DIA-NN output libraries to have proteotypic information. Don't they?
Q3: Well, depends. One can predict  any set of protein and their precursors and fragments (or starting bottom-up with another logic). This can be a subset of the whole genome of that species. Using the present species, proteins may be grouped in the library. However, I don't expect this to be very meaningful as one would want to re-group proteins on the actually observed species in a DIA experiment

**Reply by vdemichev** [reply]
FragPipe libs don't list all proteins associated with the precursor. Means it's impossible to know if the precursor is specific to a particular protein or gene. Precursors are always specific to the protein groups they've been associated to.

**Reply by JM-Bader** [reply]
"Precursors are always specific to the protein groups they've been associated to." > that's enough if the old protein grouping is kept.

**Reply by JM-Bader** [reply]
"Means it's impossible to know if the precursor is specific to a particular protein or gene". For most cases it doesn't matter if a peptide is a proteotypic peptide of that protein group or a razor peptide assigned to that group (because there is no other protein group with evidence). Or?
